src/main/java/muramasa/antimatter/network/packets/CompoundPacket.java

```diff
@@ -0,0 +1,51 @@
+package muramasa.antimatter.network.packets;
+
+import muramasa.antimatter.capability.ICapabilityHost;
+import muramasa.antimatter.util.Utils;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.world.ClientWorld;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.dimension.DimensionType;
+import net.minecraftforge.fml.network.NetworkEvent;
+
+import java.util.function.Supplier;
+
+public class CompoundPacket {
+
+    private final CompoundNBT tag;
+    private final BlockPos pos;
+    private final int dim;
+
+    public CompoundPacket(CompoundNBT tag, BlockPos pos, int dim) {
```

Comment:
Sending a full CompoundNBT is very inefficient.

Reply:
I know but it should happend only once for the initialization purpose.

Reply:
If the watcher doesn't need the information, then the initialization isn't needed. Only when the information is needed the packet should be sent, this can be done within TileEntity's packet handling or separately.

Reply:
So I can remove CompondPacket and make the packets for each handler separetely, where it needed. However, I thought that send all information will be better.

Reply:
Keep it for now, I'll work on that.